### PR TITLE
Introduce a withDefault check step

### LINKED ISSUE #4114: Core: Introduce a `withDefault` check step
Built on top of `transformOption`, see https://groups.google.com/g/gatling/c/EmBcnnApZd0/m/J80BHr3iBQAJ

### PR DESCRIPTION
**Pull request by jingcj**
Core: Introduce a withDefault check step #4114

### CODE REVIEW

File: gatling-core/src/main/scala/io/gatling/core/check/CheckBuilder.scala [hunk 1]
```diff
@@ -175,6 +176,14 @@ private final case class DefaultValidatorCheckBuilder[T, P, X](extractor: Expres
   override def transformOptionWithSession[X2](transformation: (Option[X], Session) => Validation[Option[X2]]): ValidatorCheckBuilder[T, P, X2] =
     copy(extractor = session => extractor(session).map(transformOptionExtractor(transformation(_, session))))
 
+  override def withDefault(other: Expression[X]): ValidatorCheckBuilder[T, P, X] =
+    transformOptionWithSession((actual, session) =>
+      (actual match {
+        case Some(_) => actual
+        case _       => other(session).toOption
```
Comment (slandelle) [inline gatling-core/src/main/scala/io/gatling/core/check/CheckBuilder.scala:183]: Please reverse the cases to save an unapply:

case None => ???
case _ => ???

**Review by slandelle — CHANGES_REQUESTED** [review]
Thanks!
One small nit + could you please add an entry in the documentation? https://github.com/gatling/gatling/blob/main/src/docs/content/reference/current/http/check/index.md#transforming

**Comment by slandelle** [comment]
Also, could you please sign our [CLA](https://docs.google.com/forms/d/1hbpxVqJ5hIYYJuOjOyOzub9fJFGFGvvfnB5w5LCWdUw/edit)?

File: gatling-core/src/main/scala/io/gatling/core/check/CheckBuilder.scala (hunk 1 above)
Comment (jingcj) [reply gatling-core/src/main/scala/io/gatling/core/check/CheckBuilder.scala:183]: Got it.

**Comment by jingcj** [comment]
> Also, could you please sign our [CLA](https://docs.google.com/forms/d/1hbpxVqJ5hIYYJuOjOyOzub9fJFGFGvvfnB5w5LCWdUw/edit)?

**Comment by jingcj** [comment]
> Also, could you please sign our [CLA](https://docs.google.com/forms/d/1hbpxVqJ5hIYYJuOjOyOzub9fJFGFGvvfnB5w5LCWdUw/edit)?

Sure

**Comment by slandelle** [comment]
Thanks a lot!

**Pull request closed without merging**